### PR TITLE
add ubuntu banners

### PR DESCRIPTION
**Pull request by agaurav77**
For #51 (Ubuntu)

Scraped the ChangeLog from Ubuntu's OpenSSH repo on Launchpad, some versions may repeat. (https://launchpad.net/ubuntu/+source/openssh/+changelog)

I hope this is what you intended? If you want, I can add the scraping script (also in ruby) here, for reference.

Thanks!

### CODE REVIEW

File: lib/ssh_scan/os/ubuntu.rb [hunk 1]
```diff
@@ -1,12 +1,187 @@
 module SSHScan
   module OS
     class Ubuntu
+      @@fingerprints = {
+       "warty"=> ["3.8.1p1-11ubuntu3.3", "3.8.1p1-11ubuntu3.2", "3.8.1p1-11ubuntu3"],
```
Comment (claudijd) [inline lib/ssh_scan/os/ubuntu.rb]: I think it would be more practical if the key here was an Ubuntu numbered version.

**Comment by claudijd** [comment]
Also, it would be really helpful if you included unit-tests for #os_guess method to verify the version so we know this is all working as expected.

You can see lib/ssh_scan/ssh_lib/openssh.rb as an example of how to give Ubuntu it's own version class.

**Comment by claudijd** [comment]
Also, the current data structure allows for duplicates, might also consider adding a unit-test that verifies all the strings are uniq and fail the test if they are not.

**Comment by agaurav77** [comment]
@claudijd Added the tests, and did the recommended changes. Does it look good now?

**Comment by claudijd** [comment]
@agaurav77 landed in master, nice job! thanks!

**Comment by claudijd** [comment]
Also, forgot to attach my testing verification on an Ubuntu 14.04 host.

    [
      {
        "ssh_scan_version": "0.0.10",
        "hostname": "",
        "ip": "192.168.0.36",
        "port": "22",
        "server_banner": "SSH-2.0-OpenSSH_6.6.1p1 Ubuntu-2ubuntu2.8",
        "ssh_version": 2.0,
        "os": "ubuntu",
        "os_cpe": "o:canonical:ubuntu:14.04",
        "ssh_lib": "openssh",
        "ssh_lib_cpe": "a:openssh:openssh:6.6.1p1",
        "cookie": "6eb8050eebf46ceac0c7f63f87de725c",
        "key_algorithms": [
          "curve25519-sha256@libssh.org",
          "ecdh-sha2-nistp256",
          "ecdh-sha2-nistp384",
          "ecdh-sha2-nistp521",
          "diffie-hellman-group-exchange-sha256",
          "diffie-hellman-group-exchange-sha1",
          "diffie-hellman-group14-sha1",
          "diffie-hellman-group1-sha1"
        ],
        "server_host_key_algorithms": [
          "ssh-rsa",
          "ssh-dss",
          "ecdsa-sha2-nistp256",
          "ssh-ed25519"
        ],
        "encryption_algorithms_client_to_server": [
          "aes128-ctr",
          "aes192-ctr",
          "aes256-ctr",
          "arcfour256",
          "arcfour128",
          "aes128-gcm@openssh.com",
          "aes256-gcm@openssh.com",
          "chacha20-poly1305@openssh.com",
          "aes128-cbc",
          "3des-cbc",
          "blowfish-cbc",
          "cast128-cbc",
          "aes192-cbc",
          "aes256-cbc",
          "arcfour",
          "rijndael-cbc@lysator.liu.se"
        ],
        "encryption_algorithms_server_to_client": [
          "aes128-ctr",
          "aes192-ctr",
          "aes256-ctr",
          "arcfour256",
          "arcfour128",
          "aes128-gcm@openssh.com",
          "aes256-gcm@openssh.com",
          "chacha20-poly1305@openssh.com",
          "aes128-cbc",
          "3des-cbc",
          "blowfish-cbc",
          "cast128-cbc",
          "aes192-cbc",
          "aes256-cbc",
          "arcfour",
          "rijndael-cbc@lysator.liu.se"
        ],
        "mac_algorithms_client_to_server": [
          "hmac-md5-etm@openssh.com",
          "hmac-sha1-etm@openssh.com",
          "umac-64-etm@openssh.com",
          "umac-128-etm@openssh.com",
          "hmac-sha2-256-etm@openssh.com",
          "hmac-sha2-512-etm@openssh.com",
          "hmac-ripemd160-etm@openssh.com",
          "hmac-sha1-96-etm@openssh.com",
          "hmac-md5-96-etm@openssh.com",
          "hmac-md5",
          "hmac-sha1",
          "umac-64@openssh.com",
          "umac-128@openssh.com",
          "hmac-sha2-256",
          "hmac-sha2-512",
          "hmac-ripemd160",
          "hmac-ripemd160@openssh.com",
          "hmac-sha1-96",
          "hmac-md5-96"
        ],
        "mac_algorithms_server_to_client": [
          "hmac-md5-etm@openssh.com",
          "hmac-sha1-etm@openssh.com",
          "umac-64-etm@openssh.com",
          "umac-128-etm@openssh.com",
          "hmac-sha2-256-etm@openssh.com",
          "hmac-sha2-512-etm@openssh.com",
          "hmac-ripemd160-etm@openssh.com",
          "hmac-sha1-96-etm@openssh.com",
          "hmac-md5-96-etm@openssh.com",
          "hmac-md5",
          "hmac-sha1",
          "umac-64@openssh.com",
          "umac-128@openssh.com",
          "hmac-sha2-256",
          "hmac-sha2-512",
          "hmac-ripemd160",
          "hmac-ripemd160@openssh.com",
          "hmac-sha1-96",
          "hmac-md5-96"
        ],
        "compression_algorithms_client_to_server": [
          "none",
          "zlib@openssh.com"
        ],
        "compression_algorithms_server_to_client": [
          "none",
          "zlib@openssh.com"
        ],
        "languages_client_to_server": [

        ],
        "languages_server_to_client": [

        ],
        "auth_methods": [
          "publickey",
          "password"
        ],
        "fingerprints": {
          "md5": "0e:a5:eb:8c:b5:55:7e:1f:f1:23:f3:cd:91:48:51:18",
          "sha1": "a0:1f:e5:66:60:24:31:dc:73:95:e0:0d:34:af:f5:cd:b8:75:f7:ff",
          "sha256": "96:c3:c6:2a:1e:ca:2f:63:f2:6f:9a:76:cb:52:2c:2e:65:1c:e2:cb:90:9f:22:40:0e:39:20:98:de:62:b7:3a"
        },
        "compliance": {
          "policy": "Mozilla Modern",
          "compliant": false,
          "recommendations": [
            "Remove these Key Exchange Algos: diffie-hellman-group-exchange-sha1, diffie-hellman-group14-sha1, diffie-hellman-group1-sha1",
            "Remove these MAC Algos: hmac-md5-etm@openssh.com, hmac-sha1-etm@openssh.com, umac-64-etm@openssh.com, hmac-ripemd160-etm@openssh.com, hmac-sha1-96-etm@openssh.com, hmac-md5-96-etm@openssh.com, hmac-md5, hmac-sha1, umac-64@openssh.com, hmac-ripemd160, hmac-ripemd160@openssh.com, hmac-sha1-96, hmac-md5-96",
            "Remove these Encryption Ciphers: arcfour256, arcfour128, aes128-cbc, 3des-cbc, blowfish-cbc, cast128-cbc, aes192-cbc, aes256-cbc, arcfour, rijndael-cbc@lysator.liu.se"
          ],
          "references": [
            "https://wiki.mozilla.org/Security/Guidelines/OpenSSH"
          ]
        }
      }
    ]

**Pull request merged**